diff --git a/src/idevicerestore.c b/src/idevicerestore.c
index b0572ca..b0b6a9e 100644
--- a/src/idevicerestore.c
+++ b/src/idevicerestore.c
@@ -1033,6 +1033,11 @@ int idevicerestore_start(struct idevicerestore_client_t* client)
 	}
 
 	idevicerestore_progress(client, RESTORE_STEP_PREPARE, 0.2);
+	if (client->mode == MODE_RESTORE) {
+		if (client->flags & FLAG_ALLOW_RESTORE_MODE) {
+			tss_enabled = 0;
+		}
+	}
 
 	/* retrieve shsh blobs if required */
 	if (tss_enabled) {
diff --git a/src/restore.c b/src/restore.c
index 257cc38..862bb75 100644
--- a/src/restore.c
+++ b/src/restore.c
@@ -302,6 +302,15 @@ irecv_device_t restore_get_irecv_device(struct idevicerestore_client_t* client)
 	}
 
 	plist_get_string_val(node, &model);
+	fprintf(stderr, "%s: Found model %s\n", __func__, model);
+
+	/* HACK: Replace DEV devices with AP equivalent devices
+	 * So that it can enjoy FDR-less restore
+	 */
+	if (strstr(model, "DEV")) {
+		strncpy(strstr(model, "DEV"), "AP\0", 3);
+	}
+
 	irecv_devices_get_device_by_hardware_model(model, &irecv_device);
 	free(model);
 
